gnmi_cli_py/py_gnmicli.py

```diff
@@ -465,46 +470,69 @@ def main():
-  elif mode == 'subscribe':
-    request_iterator = gen_request(paths, args, prefix)
-    subscribe_start(stub, args, request_iterator)
+
+  if create_connections < -1:
+    print('''
+          Default number of TCP connections with gNMI server is 1.
+          Please use the '--create_connections <positive_number>' to
+          create TCP connections or use '--create_connections -1' to
+          create infinite TCP connections.
+          ''')
+    sys.exit(1)
+
+  while True:
+    if create_connections > 0:
+        create_connections -= 1
+    elif create_connections == 0:
+        break
+
+    try:
+      stub = _create_stub(creds, target, port, host_override)
+      if mode == 'get':
+        print('Performing GetRequest, encoding=JSON_IETF', 'to', target,
+              ' with the following gNMI Path\n', '-'*25, '\n', paths)
+        response = _get(stub, paths, user, password, prefix)
+        print('The GetResponse is below\n' + '-'*25 + '\n')
+        if form == 'protobuff':
+          print(response)
+        elif response.notification[0].update[0].val.json_ietf_val:
+          print(json.dumps(json.loads(response.notification[0].update[0].val.
+                                      json_ietf_val), indent=2))
+        elif response.notification[0].update[0].val.string_val:
+          print(response.notification[0].update[0].val.string_val)
+        else:
+          print('JSON Format specified, but gNMI Response was not json_ietf_val')
+          print(response)
+      elif mode == 'set-update':
+        print('Performing SetRequest Update, encoding=JSON_IETF', ' to ', target,
+              ' with the following gNMI Path\n', '-'*25, '\n', paths, json_value)
+        response = _set(stub, paths, 'update', user, password, json_value)
+        print('The SetRequest response is below\n' + '-'*25 + '\n', response)
+      elif mode == 'set-replace':
+        print('Performing SetRequest Replace, encoding=JSON_IETF', ' to ', target,
+              ' with the following gNMI Path\n', '-'*25, '\n', paths)
+        response = _set(stub, paths, 'replace', user, password, json_value)
+        print('The SetRequest response is below\n' + '-'*25 + '\n', response)
+      elif mode == 'set-delete':
+        print('Performing SetRequest Delete, encoding=JSON_IETF', ' to ', target,
+              ' with the following gNMI Path\n', '-'*25, '\n', paths)
+        response = _set(stub, paths, 'delete', user, password, json_value)
+        print('The SetRequest response is below\n' + '-'*25 + '\n', response)
+      elif mode == 'subscribe':
+        request_iterator = gen_request(paths, args, prefix)
+        subscribe_start(stub, args, request_iterator)
+    except grpc.RpcError as err:
+      if err.code() == grpc.StatusCode.UNAVAILABLE:
+        print("Client receives an exception '{}' indicating gNMI server is shut down and Exiting ..."
+              .format(err.details()))
+        sys.exit(2)
```

Comment:
>2 [](http://example.com/codeflow?start=17&length=1)

Since you use multiple return code to indicate different failure reasons. Better to define constants.

Reply:
Updated.